lib/model/rofolder.go

```diff
@@ -16,113 +16,135 @@ import (
 
 type roFolder struct {
 	stateTracker
+	scan
 
-	folder    string
-	intv      time.Duration
-	timer     *time.Timer
-	model     *Model
-	stop      chan struct{}
-	scanNow   chan rescanRequest
-	delayScan chan time.Duration
+	folderId string
+	model    *Model
+	stop     chan struct{}
 }
 
 type rescanRequest struct {
-	subs []string
-	err  chan error
+	subdirs []string
+	err     chan error
 }
 
-func newROFolder(model *Model, folder string, interval time.Duration) *roFolder {
+// bundle all folder scan activity
+type scan struct {
+	scanInterval time.Duration
```

Comment:
I'd like to prevent code lines like (imaginary)

``` 
f.timer = ....
f.interval = ....
```

We have some long methods (>20lines) and this adds some complexity in readibility.